src/repo.rs

```diff
@@ -269,6 +269,34 @@ impl Repo {
         Ok(self.resolve_rev(refspec, false)?.unwrap())
     }
 
+    /// Query metadata for a content object.
+    /// This is similar to [`load_file`], but is more efficient if reading the file content is not needed.
+    pub fn query_file<P: IsA<gio::Cancellable>>(
+        &self,
+        checksum: &str,
+        cancellable: Option<&P>,
+    ) -> Result<(gio::FileInfo, glib::Variant), glib::Error> {
+        unsafe {
+            let mut out_file_info = ptr::null_mut();
+            let mut out_xattrs = ptr::null_mut();
+            let mut error = ptr::null_mut();
+            let _ = ffi::ostree_repo_load_file(
```

Comment:
Minor: it would be good to check/assert the returned boolean.

Reply:
For now I added `debug_assert!` checks.